--- firefox-52.9.0esr/gfx/2d/BaseRect.h.orig	2017-04-11 04:13:04.000000000 +0200
+++ firefox-52.9.0esr/gfx/2d/BaseRect.h	2024-08-24 19:58:34.282296752 +0200
@@ -8,6 +8,8 @@
 
 #include <algorithm>
 #include <cmath>
+#include <cstdint>
+#include <limits>
 #include <ostream>
 
 #include "mozilla/Assertions.h"
--- firefox-52.9.0esr/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/opus/audio_encoder_opus.cc.orig	2017-04-11 04:13:15.000000000 +0200
+++ firefox-52.9.0esr/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/opus/audio_encoder_opus.cc	2024-08-24 21:26:32.327036464 +0200
@@ -8,6 +8,9 @@
  *  be found in the AUTHORS file in the root of the source tree.
  */
 
+#include <cstdint>
+#include <limits>
+
 #include "webrtc/modules/audio_coding/codecs/opus/interface/audio_encoder_opus.h"
 
 #include "webrtc/base/checks.h"
