--- firefox-52.9.0esr/tools/profiler/core/platform.h~	2020-03-03 13:20:04.000000000 +0100
+++ firefox-52.9.0esr/tools/profiler/core/platform.h	2020-03-03 13:20:47.438599518 +0100
@@ -63,6 +63,7 @@
 // provide a wrapper for it (except for Bionic)
 #if defined(__linux__)
 #include <unistd.h>
+#if !__GLIBC_PREREQ(2, 30)
 #if !defined(__BIONIC__)
 #include <sys/syscall.h>
 static inline pid_t gettid()
@@ -71,6 +72,7 @@ static inline pid_t gettid()
 }
 #endif
 #endif
+#endif
 
 #ifdef XP_WIN
 #include <windows.h>
--- firefox-52.9.0esr/memory/mozjemalloc/jemalloc.c.orig	2017-05-12 02:25:50.000000000 +0200
+++ firefox-52.9.0esr/memory/mozjemalloc/jemalloc.c	2023-01-30 06:15:52.731524546 +0100
@@ -331,7 +331,7 @@ __FBSDID("$FreeBSD: head/lib/libc/stdlib
 #endif
 #include <sys/time.h>
 #include <sys/types.h>
-#if !defined(MOZ_MEMORY_SOLARIS) && !defined(MOZ_MEMORY_ANDROID)
+#if !defined(MOZ_MEMORY_SOLARIS) && !defined(MOZ_MEMORY_ANDROID) && !defined(__linux__)
 #include <sys/sysctl.h>
 #endif
 #include <sys/uio.h>
